### PR TITLE
[ci] Use Jenkins

### PR DESCRIPTION
**Pull request by jgsogo**
Trying Jenkins approach, first draft. Something like this:

image

### CODE REVIEW

File: modern/.env [hunk 1]
```diff
@@ -1,4 +1,5 @@
 CONAN_VERSION=1.37.2
+DOCKER_TAG=1.37.2
```
Comment (uilianries) [inline modern/.env:2]: Why separate DOCKER_TAG and CONAN_VERSION?

File: .ci/xenial.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,148 @@
+// These are things to parameterize/read/write from/to .env file
+List gccVersions = ['5', '6', '7', '8', '9', '10', '11']
+List clangVersions = ['10', '11', '12']
+boolean isMaster = (scm.branches[0].name == 'master')
+echo ">> scm.branches[0].name: ${scm.branches[0].name}"
+List<String> environment = []
+
+def configure_docker() {
+    withCredentials([usernamePassword(credentialsId: 'c3istg-dad-docker', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
```
Comment (uilianries) [inline .ci/xenial.jenkinsfile:9]:
```suggestion
    withCredentials([usernamePassword(credentialsId: 'c3istg-dad-docker', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
```

What about the official hub.docker registry? People will download from there, at least, we could upload a copye. I know using our side could save CI time as all is at same server, but still don't want to configure a new registry only to download a single image.

Comment (jgsogo) [reply .ci/xenial.jenkinsfile:9]: Sure, PRs should be cached on our side, when the PR is merged to `master` then those images will be built again (or just promoted) to DockerHub.

File: modern/.env (hunk 1 above)
Comment (jgsogo) [reply modern/.env:2]: I need to tag the images built from PRs with the corresponding commit, so they are not used by other jobs running other branches/pull requests. Only images generated in `master` can use the tag with the Conan version.

**Comment by jgsogo** [comment]
Green and building all the images. Total time 1h 41min without using cached images.

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**